cg/meta/orders/ticket_handler.py

```diff
@@ -46,16 +49,20 @@ def create_ticket(
             project=project,
         )
         attachment: dict = self.create_attachment(order=order)
-        ticket_nr: str | None = self.osticket.open_ticket(
-            name=user_name,
+
+        # Create ticket in Freshdesk
+        freshdesk_ticket = TicketCreate(
```

Comment:
The encapsulation of the fresh desk logic is leaky.

We don't want to have logic mentioning fresh desk spread through out our code base. The usage of the client needs to be wrapped in a service. But I guess this `TicketHandler` is the ticketing service.

Ideally it would look something like this.

```python
class TicketingService:
    def __init__(self, client: FreshDeskAPIClient):
        self.client = client
    
    def create_ticket(self, email: str, name: str, subject: str, attachment: dict) -> int:
        ticket = TicketCreate(
            email=email,
            name=name,
            subject=subject,
            attachment=[attachment]
        )
        ticket_response: TicketResponse = self.freshdesk_client.create_ticket(ticket)
        response = self.client.create_ticket(ticket)
        LOG.info(f"{ticket_response.id}: opened new ticket in Freshdesk")
        return ticket_response.id
```

Reply:
But looking at these changes it seems like you want to ensure that fresh desk tickets are created in parallel which is fine.

Feel free to disregard this for now, it is a bigger refactoring of existing code.

Reply:
I agree with SA. I think it should be referred to as client in the TicketHandler, and maybe ticketing_client in the OrdersAPI.

Reply:
Thanks for the clarifications. Will consider that when we remove `osticket` !